### PR TITLE
Proposal: kafka output

### PR DESCRIPTION
**Pull request by urso**
Kafka output basically wrapping github.com/Shopify/sarama library in libbeat + make it configurable.

Depends on: #927 and #940

### CODE REVIEW

File: libbeat/outputs/kafka/kafka.go [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+	}
+
+	hosts := config.Hosts
+	if len(hosts) < 1 {
+		logp.Err("Kafka configuration failed with: %v", errNoHosts)
+		return errNoHosts
+	}
+	debugf("hosts: %v", hosts)
+
+	useType := kafkaDefaultTopicsFromType
```
Comment (tsg) [inline libbeat/outputs/kafka/kafka.go]: I think this configuration option is not documented.

Comment (urso) [inline libbeat/outputs/kafka/kafka.go]: oh yes, forgot this one.

File: libbeat/docs/outputconfig.asciidoc [hunk 2]
```diff
@@ -523,6 +524,79 @@ setting does not affect how events are published.
...
+
+===== flush_interval
+
+The number of seconds to wait for new events between two producer API calls.
+
+===== tls
+
+Configuration options for TLS parameters like the certificate authority to use
+for HTTPS-based connections. If the `tls` section is missing, the host CAs are used for HTTPS connections to
+Elasticsearch.
```
Comment (tsg) [inline libbeat/docs/outputconfig.asciidoc]: s/Elasticsearch/Kafka? Do all the TLS parameters work for Kafka?

Comment (urso) [inline libbeat/docs/outputconfig.asciidoc]: ups and yes

**Comment by tsg** [comment]
LGTM. I suggest some integration tests for a few of the common configuration (TLS/no-TLS, ACK/no-ACK), but those can come in a different PR.

**Comment by urso** [comment]
History cleanup finished

**Comment by biswajit86** [comment]
Is there a need for implementing other PR's or can we start using kafka output in beats framework.

**Comment by anefassa** [comment]
This is great news on providing a Kafka output for beats. When is v5.0.0-GA targeted to be released? 

In the meantime I have built from master to start testing it out. Any tips on how to configure the Kafka output?

**Comment by andrewkroh** [comment]
Docs: https://www.elastic.co/guide/en/beats/filebeat/master/kafka-output.html
And be aware of this: https://github.com/elastic/beats/issues/1215#issuecomment-200554532

**Comment by anefassa** [comment]
Thanks for sharing the docs. Kafka output with filebeat is working for me.

**Comment by mostolog** [comment]
Could anyone @elastic confirms if beat -> kafka is going to be available for version 5?

**Comment by ruflin** [comment]
Yes, Kafka will be available in the 5.0 release. You can already tests it with the 5.0.0-alpha releases.

**Pull request merged**